### ISSUE TITLE
opencog/atomspace tries to install scheme files to  /usr/share/guile/site/2.2/opencog/base

### ISSUE THREAD

**Issue by vsbogd** [opencog/atomspace#1968]
Recently opencog started to install some files into  `/usr/share/guile/site/2.2/opencog/base`. Changing prefix via `CMAKE_INSTALL_PREFIX` doesn't work.
```
Install the project...
-- Install configuration: "Debug"
-- Installing: /home/vital/projects/opencog/builds/current/lib/opencog/libatom_types.so
-- Set runtime path of "/home/vital/projects/opencog/builds/current/lib/opencog/libatom_types.so" to "/home/vital/projects/opencog/builds/current/lib/opencog:lib/python3.6/dist-packages/opencog:/home/vital/projects/opencog/builds/current/lib"
-- Installing: /home/vital/projects/opencog/builds/current/include/opencog/atoms/atom_types/atom_types.h
CMake Error at opencog/atoms/atom_types/cmake_install.cmake:78 (file):
  file cannot create directory: /usr/share/guile/site/2.2/opencog/base.
  Maybe need administrative privileges.
Call Stack (most recent call first):
  opencog/atoms/cmake_install.cmake:42 (include)
  opencog/cmake_install.cmake:42 (include)
  cmake_install.cmake:43 (include)
```

**Comment by linas** [comment]
Huh. Yeah, that should probably be `CMAKE_INSTALL_PREFIX/share/guile/site/2.2`

**Comment by linas** [comment]
So is this an issue or can it be closed?

**Comment by vsbogd** [comment]
My point of view it is a guile issue, as if guile was installed from package is has only `/usr/share` paths in `%load-path`, so one cannot install new modules into `/usr/local`.

I think the fix is to allow set guile site-directory as `cmake` variable.

**Comment by vsbogd** [comment]
raised https://github.com/opencog/atomspace/pull/2044 to fix

**Comment by AmeBel** [comment]
@vsbogd why would you want to install in `/usr/local/` if guile expects the modules in `/usr/share/`? Or am I missing something?

**Comment by vsbogd** [comment]
@AmeBel , it is not about `/usr/local` only. I have raised the issue because I was building `atomspace` using `CMAKE_INSTALL_PREFIX` to override prefix. My `CMAKE_INSTALL_PREFIX` points to the my folder which I can modify without using root permissions. So I execute just `make install` without `sudo`. And at the moment this fails because of Guile install path.

May be I am not right about guile, but my understanding that `/usr/local` is common way to install self-compiled binaries and it is strange that Guile doesn't support it. Anyway using single prefix to override all of installation paths of the software being compiled is very common and I think we should support it.

**Comment by AmeBel** [comment]
@vsbogd Got it thanks

**Comment by linas** [comment]
closing; this appears to be fixed in #2044
